### DISCUSSION TITLE
Can't seem to get response from external API with getStaticProps

### DISCUSSION THREAD

**Discussion by 1898182** [vercel/next.js#21728] [Help] [answered]
I am trying to use getStaticProps for the first time so I am doing a simple axios call using [dummyapi.io](https://dummyapi.io) , but it returns `undefined`. I even copied an example from this [blog](https://medium.com/swlh/fetching-and-hydrating-a-next-js-app-using-getserversideprops-and-getstaticprops-65bfe42afed8) but it doesn't seem to work (check last code snippet at the bottom of the page).

If I do the axios call inside useEffect, data is being returned. Any pointers?

```
import axios from 'api/client' // defined baseURL and my API_ID in headers

const getUsers = async () => await axios.get('/user').then(res => {data: res.data})

const myComponent = ({ data }) => {
     useEffect(() => {
       console.log(data) // undefined
     }

   return ( 
       ... 
   )
}

export const getStaticProps = async () => {
  const data = await getUsers()

  return {
    props: {
      data: data
    }
  }
}
```

**Comment by benweier** [comment, marked as answer]
Your `getUsers` function is actually returning undefined.

```
res => {data: res.data}
```
should be
```
res => ({data: res.data}))
```

**Reply by 1898182** [reply]
Is there a reason that I don't know why I SHOULD be wrapping it with parentheses? To my knowledge, I don't need them because it's just syntax sugar for ES6 anonymous function, just like I could wrap the argument `(res) => object_to_return`. I still tested it though, but it returns undefined again.

**Reply by benweier** [reply]
The reason is that without parenthesis the object brace looks like a normal function block

```
const callback = (res) => {
  res.data
  // no explicit return statement
}
```

With parenthesis the JS engine treats it as an implicit return statement
```
const callback = (res) => ({
  data: res.data
})
```

This "rule" only applies to objects due to the common character. You don't need the parenthesis when returning a variable (even if that variable is an object) or string/number/boolean, even arrays
```
const callback = (res) => res.data
```
```
const callback = (res) => 'string' || 1 || true
```
```
const callback = (res) => ['string']
```

**Reply by benweier** [reply]
Your component name makes me want to check that you're exporting `getStaticProps` from a page module (`pages/users.js`) and not a component module (`components/users.js`)

The `useEffect` will work in a component (as well as a page) but only top level pages allow `getStaticProps`

**Reply by 1898182** [reply]
Thank you for your explanation. That makes very much sense. Since I wrote it in a single line, I didn't see it like a function. Also I did not put the whole code since **it is** a page component. `pages/homepage/index.jsx`.

Your second comment made me realize I have my page wrapped in a Layout component, which would mean then my page `homepage` is not a the top level. I have tried under that root page and it is working. 

If I move the request in a true component – like a provider perhaps – I could use it anywhere?

Thank you kind sir! 🦸‍♂️
